File: client/src/Components/Schedule/SortSchedule.js
export default function SortSchedule(schedule) {
  schedule.sort((schedule1, schedule2) => {
    if (schedule1.number_pair > schedule2.number_pair) return 1;
    else if (schedule1.number_pair === schedule2.number_pair) {
      if (schedule1.pair_type > schedule2.pair_type) return 1;
      else if (schedule1.pair_type === schedule2.pair_type) {
        if (schedule1.day_week > schedule2.day_week) return 1;
        else if (schedule1.day_week === schedule2.day_week) {
          return 0;
        } else return -1;
      } else return -1;
    } else return -1;
  });
}

File: server/Algorithms/EvalutionAlgorithmCpp/EvalutionAlgorithmCpp/SortSchedule.cpp
#include "SortSchedule.h"
#include <iostream>
using namespace std;
bool Compare(schedule a, schedule b);
void SortSchedule(vector<schedule>& i_schedule){
    int right = i_schedule.size()-1;
    while(right>0){
        for(int j =0;j<right;j++){
            if(Compare(i_schedule[j], i_schedule[j+1])){
                schedule temp = i_schedule[j+1];
                i_schedule[j+1]= i_schedule[j];
                i_schedule[j]=temp;
            }
        }
        right--;
    }
}
bool Compare(schedule a, schedule b){
    if (a.day_week > b.day_week) return true;
    if (a.day_week < b.day_week) return false;
    if (a.number_pair > b.number_pair) return true;
    if (a.number_pair < b.number_pair) return false;
    if (a.pair_type > b.pair_type) return true;
    if (a.pair_type < b.pair_type) return false;
    return false;
}
